src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/HubConnectionTest.java

```diff
@@ -2565,6 +2565,31 @@ public void sendWithCustomObjectTriggersOnHandlerWithMessagePack()  {
         assertEquals((short) 5, (short) person.getT());
     }
 
+    @Test
+    public void throwFromOnHandlerRunsAllHandlers() {
+        AtomicReference<String> value1 = new AtomicReference<>();
+        AtomicReference<String> value2 = new AtomicReference<>();
+
+        MockTransport mockTransport = new MockTransport();
+        HubConnection hubConnection = TestUtils.createHubConnection("http://example.com", mockTransport);
+
+        hubConnection.on("inc", (param1) -> {
+            value1.set(param1);
+            throw new RuntimeException("throw from on handler");
```

Comment:
Is there no way to verify the exception details got logged?

Reply:
I think we have a test logger that I can inspect and assert the logs